HLTrigger/JSONMonitoring/plugins/L1TriggerJSONMonitoring.cc

```diff
@@ -0,0 +1,452 @@
+  }
+  
+  L1GlobalType_.clear();   
+  L1Global_.clear();     
+  
+  //Set the experimentType -          
+  L1GlobalType_.push_back( "Physics" );
+  L1GlobalType_.push_back( "Calibration" );
+  L1GlobalType_.push_back( "Random" );
+
+  const unsigned int la = L1AlgoNames_.size();       
+  const unsigned int lg = L1GlobalType_.size();      
+  
+  //Resize per-path counters   
+  L1AlgoAccept_.resize(la);         
+  L1AlgoAcceptPhysics_.resize(la);         
+  L1AlgoAcceptCalibration_.resize(la);         
+  L1AlgoAcceptRandom_.resize(la);         
+  
+  L1Global_.resize(lg);                 
+  
+  resetLumi();
+
+  //Write the once-per-run files if not already written
+  //Eventually must rewrite this with proper multithreading (i.e. globalBeginRun)
+  bool expected = false;
+  if( runCache()->wroteFiles.compare_exchange_strong(expected, true) ){
+    runCache()->wroteFiles = true;
+
+    unsigned int nRun = iRun.run();
+        
+    //Create definition file for L1 Rates -  
+    std::stringstream ssL1Jsd;
+    ssL1Jsd << "run" << std::setfill('0') << std::setw(6) << nRun << "_ls0000";
+    ssL1Jsd << "_streamL1Rates_pid" << std::setfill('0') << std::setw(5) << getpid() << ".jsd";
+    stL1Jsd_ = ssL1Jsd.str();
+
+    writeL1DefJson(baseRunDir_ + "/" + stL1Jsd_);
+    
+    //Write ini files    
+    //L1
+    Json::Value l1Ini;
+    Json::StyledWriter writer;
+
+    Json::Value l1AlgoNamesVal(Json::arrayValue);
+    for (unsigned int ui = 0; ui < L1AlgoNames_.size(); ui++){
+      l1AlgoNamesVal.append(L1AlgoNames_.at(ui));
+    }
+
+    Json::Value eventTypeVal(Json::arrayValue);
+    for (unsigned int ui = 0; ui < L1GlobalType_.size(); ui++){
+      eventTypeVal.append(L1GlobalType_.at(ui));
+    }
+
+    l1Ini["L1-Algo-Names"] = l1AlgoNamesVal;
+    l1Ini["Event-Type"]    = eventTypeVal;
+    
+    std::string && result = writer.write(l1Ini);
+  
+    std::stringstream ssL1Ini;
+    ssL1Ini << "run" << std::setfill('0') << std::setw(6) << nRun << "_ls0000_streamL1Rates_pid" << std::setfill('0') << std::setw(5) << getpid() << ".ini";
+    
+    std::ofstream outL1Ini( monPath + ssL1Ini.str() );
+    outL1Ini<<result;
+    outL1Ini.close();
+  }
+
+  //Initialize variables for verification of prescaleIndex
+  oldLumi          = 0;
+  oldPrescaleIndex = 100;
+
+}//End beginRun function        
+
+void L1TriggerJSONMonitoring::beginLuminosityBlock(const edm::LuminosityBlock&, const edm::EventSetup& iSetup){ resetLumi(); }
+
+std::shared_ptr<l1Json::lumiVars>
+L1TriggerJSONMonitoring::globalBeginLuminosityBlockSummary(const edm::LuminosityBlock& iLumi, const edm::EventSetup& iSetup, const LuminosityBlockContext* iContext)
+{
+  std::shared_ptr<l1Json::lumiVars> iSummary(new l1Json::lumiVars);
+
+  unsigned int MAXPATHS = 512;
+
+  iSummary->processed = new HistoJ<unsigned int>(1, 1);
+
+  iSummary->prescaleIndex = 100;
+
+  iSummary->L1AlgoAccept            = new HistoJ<unsigned int>(1, MAXPATHS);                            
```

Comment:
thanks - they should certainly also be reinitialized to nullptr. Please add a new PR for that. Thanks.